include/proxy-wasm/context.h

```diff
@@ -106,215 +102,192 @@ class ContextBase {
   // Called before deleting the context.
   virtual void destroy();
 
-  // VM level downcalls into the WASM code on Context(id == 0).
-  virtual bool onStart(std::shared_ptr<PluginBase> plugin);
-  virtual bool onConfigure(std::shared_ptr<PluginBase> plugin);
+  /**
+   * Calls into the VM.
+   * These are implemented by the proxy-independent host code. They are virtual to support some
+   * types of testing.
+   */
 
-  // Root Context downcalls into the WASM code Context(id != 0, root_context_id_ == 0);
-  virtual void onTick();
+  // Context
+  void onCreate(uint32_t parent_context_id) override;
+  bool onDone() override;
+  void onFinalized() override;
```

Comment:
As far as I recall, the suggestion was to merge `onLog` into `onDone`, and not just rename it.

Reply:
We can't merge it with onDone.   These two calls occur at different times. onDone occurs when the connection closes.  onFinalize occurs after all the filters have processed the onDone callback in order and perhaps have updated the metadata in the FilterState.  This allows onFinalize to use any information required from filters which run later in the logs.   In a sense we are making two passes in order to allow communication both up and down the filter stack.

Reply:
We discussed this offline.  This is a core lifecycle change and invasive to current tests.  Let's get this into the spec (along with the lifecycle) and get it signed off then I'll update all the repos to match the spec.  I'll backout the name change onLog->onFinalized from these PRs.  These are superficial PRs and were not meant to include core ABI and lifecycle changes in any case.

Reply:
Done. PTAL